### ISSUE TITLE
Issues with the python binding

### ISSUE THREAD

**Issue by sphaero** [zeromq/czmq#1017]
```
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    from zyre import Zyre, ZyreEvent
  File "/home/people/arnaud/src/zyre/bindings/python/zyre.py", line 10, in <module>
    import czmq
  File "/home/people/arnaud/src/czmq/bindings/python/czmq.py", line 732
    def from(string):
```
The 'from' word is reserved in Python. Renaming it solves it but this must be done in the generator?

**Comment by Asmod4n** [comment]
@hintjens maybe a list of reserved keywords for the generator would help, it could then quit with a error message so something like this cannot happen?

**Comment by sphaero** [comment]
I think we need to add some logic to prevent python reserved words being used in the binding:
https://docs.python.org/2/library/keyword.html
But then that must be done in GSL

**Comment by taotetek** [comment]
Example - https://github.com/taotetek/tailz/blob/master/bindings/python/tailz.py#L86

**Comment by jemc** [comment]
In the QML bindings, I ended up doing some translation of reserved words to avoid language conflicts.  Specifically, `new` and `destroy` and reserved words in QML, so I substituted them to `construct` and `destruct`, which are a little more idiomatic.
https://github.com/zeromq/zproject/blob/e836ce9afac2c3d5ea88d41c4abe573fb4ac0627/zproject_bindings_qml.gsl#L141-L148

I don't think aborting the code generation is the best solution - as I doubt the binding language constraints should effectively dictate API design decisions for the main library.  Instead, I think we need to build language-specific workarounds or name "mangling" into the generators based on specific problems that projects run into.

Maybe in python, we could append a trailing underscore if the name is in the list of reserved words?

**Comment by twhittock** [comment]
There is some support for this already I believe - I think it handles the
'is' keyword already in exactly the way you suggest.

**Comment by Asmod4n** [comment]
Sounds good :)

**Comment by jemc** [comment]
As @twhittock mentioned, the `is` keyword is handled by appending an underscore: https://github.com/zeromq/zproject/blob/e836ce9afac2c3d5ea88d41c4abe573fb4ac0627/zproject_bindings_python.gsl#L116-L118

If someone wanted to patch this to support more keywords, it should be fairly simple.

**Comment by twhittock** [comment]
Fixed in #1051

**Comment by jemc** [comment]
Thanks for taking care of this!
